### DISCUSSION TITLE
Improve changelog

### DISCUSSION THREAD

**Discussion by st-pasha** [flame-engine/flame#1139] [Ideas]
I suggest that the changelog can be improved if each entry included a reference to a relevant issue/PR.

For example, instead of
```
 - `changePriority` no longer breaks game loop iteration
 - Move component mixin checks to their own files
 - Fix exception when game was rebuilt
 - Add `@mustCallSuper` on `Component.render`
```
we would say
```
 - `changePriority` no longer breaks game loop iteration (#998)
 - Move component mixin checks to their own files (#992)
 - Fix exception when game was rebuilt (#1003)
 - Add `@mustCallSuper` on `Component.render` (#1011)
```

When viewing the Changelog, Github will render these as links, so that if anyone wants to learn about each particular change in more details, they could simply follow the link.

**Comment by OlegNovosad** [comment]
Agree on that! Does it make sense to also include contributors? In such case it can give more boost in motivation maybe. Usually other products do this.

**Comment by spydon** [comment]
We are moving to Melos once v1 is released, in Melos the changelog is generated from the PRs instead of a file that you manually edit.

**Reply by spydon** [reply]
Here is an example of how it will look:
https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/CHANGELOG.md

**Reply by Salakar** [reply]
Hey folks 👋 wanted to give you a heads up that I'm working on finalising and shipping Melos v1 over the next couple of days

**Reply by spydon** [reply]
![image](https://user-images.githubusercontent.com/744771/145277250-8aa7e84d-5414-4f02-a74c-7ac1c349f2fd.png)
Earlier today :point_up: :joy: 
That's great to hear @Salakar!

**Reply by Salakar** [reply]
https://pub.dev/packages/melos/versions/1.0.0

**Reply by Salakar** [reply]
Hey folks, lmk if you want a hand with your first melos powered release since I'm still working on docs for adopting melos versioning/releases for existing repos, but the main TLDR is;

For your very first use of `melos version` use the `--since` flag and the commit sha of your last commit relating to the previous release (e.g. https://github.com/flame-engine/flame/commit/b6ddae5c61c6e59a68f0f9f3282ab32438dad916) and melos will start from there. 

You may need to keep using the since flag until all your packages have had a release through melos as your current git tags are not compatible, melos uses the format used by flutter and official plugins `<packageName>-v<version>` (yours does not have the 'v' it seems, maybe melos should allow other formats here also 🤔). Alternatively if you create new tags in that format for all your existing versions on that commit sha I linked earlier then melos should be able to go from there without needing to use since flag.

Please let me know if there's anything you need or that is blocking you.

Also check out the FlutterFire release process docs since they cover releases using melos including how to review changelogs manually should you need to (similar docs will be added to official melos docs).

**Reply by st-pasha** [reply]
Is there a way to automatically filter out certain PRs from the changelog? For example, if we don't want to include anything tagged as `chore:`, `build:` or `refactor:`?

**Reply by spydon** [reply]
> Is there a way to automatically filter out certain PRs from the changelog? For example, if we don't want to include anything tagged as `chore:`, `build:` or `refactor:`?

I think we should include everything, the changelog is for us too. :)

**Reply by spydon** [reply]
> Hey folks, lmk if you want a hand with your first melos powered release since I'm still working on docs for adopting melos versioning/releases for existing repos, but the main TLDR is;
> 
> For your very first use of `melos version` use the `--since` flag and the commit sha of your last commit relating to the previous release (e.g. https://github.com/flame-engine/flame/commit/b6ddae5c61c6e59a68f0f9f3282ab32438dad916) and melos will start from there. 

Sounds easy enough!

> You may need to keep using the since flag until all your packages have had a release through melos as your current git tags are not compatible, melos uses the format used by flutter and official plugins `<packageName>-v<version>` (yours does not have the 'v' it seems, maybe melos should allow other formats here also 🤔). Alternatively if you create new tags in that format for all your existing versions on that commit sha I linked earlier then melos should be able to go from there without needing to use since flag.

I think we'd be fine with a new tag format, not sure why we don't have a `v`. 

> Please let me know if there's anything you need or that is blocking you.

Can we choose the version number ourselves? We are going to be more flutter style in versioning than semver.

Thanks for being so supportive! 😊
